### PR TITLE
docs(blade): added documentation for tabs

### PR DESCRIPTION
**Pull request by anuraghazra**
- [PR 1](https://github.com/razorpay/blade/pull/1694) - Web Implementation
- [PR 2](https://github.com/razorpay/blade/pull/1707) - Native Implementation
- [PR 3](https://github.com/razorpay/blade/pull/1715) - Documentation (this PR)

### CODE REVIEW

File: packages/blade/src/components/Tabs/TabItem.web.tsx [hunk 1]
```diff
@@ -149,14 +149,17 @@ const TabItem = ({
...
           })}
         >
-          {leading
-            ? React.cloneElement(leading as React.ReactElement, {
-                size: iconSizeMap[size!],
-                color: isSelected
+          {Leading ? (
+            <Leading
+              size={iconSizeMap[size!]}
+              // @ts-expect-error TODO: icons types don't support brand colors
```
Comment (saurabhdaware) [inline packages/blade/src/components/Tabs/TabItem.web.tsx:155]: What is the todo action here? to add brand colors on icon?

or should this be internal usecase?

File: packages/blade/src/components/Tabs/TabList.web.tsx [hunk 2]
```diff
@@ -30,7 +35,7 @@ const TabList = ({ children }: { children: React.ReactNode }): React.ReactElemen
   }, [children, selectedValue, setSelectedValue]);
 
   return (
-    <BaseBox display={isVertical ? 'flex' : 'block'} flexShrink={0} overflow="hidden">
+    <Box {...props} display={isVertical ? 'flex' : 'block'} flexShrink={0} overflow="hidden">
```
Comment (saurabhdaware) [inline packages/blade/src/components/Tabs/TabList.web.tsx:38]: expected to support all BoxProps here? or do we want to support styled-props only?

**Review by saurabhdaware — COMMENTED** [review]
Documentation looks good to me. 

Added some comments on other changes

File: packages/blade/src/components/Tabs/TabList.web.tsx (hunk 2 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Tabs/TabList.web.tsx:38]: Also is same change needed in .native file too?

File: packages/blade/src/components/Tabs/TabItem.web.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Tabs/TabItem.web.tsx:155]: We can also expose the brand colors, will discuss this with designers once.

File: packages/blade/src/components/Tabs/TabList.web.tsx (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/src/components/Tabs/TabList.web.tsx:38]: I think we can pass the styled props instead.

File: packages/blade/src/components/Tabs/TabItem.web.tsx (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Tabs/TabItem.web.tsx:155]: Cool. Maybe just create a github issue for now to keep track, we can discuss later. Not a blocker.

Comment (anuraghazra) [reply packages/blade/src/components/Tabs/TabItem.web.tsx:155]: Oh wait I think we already added this https://github.com/razorpay/blade/pull/1678

File: packages/blade/src/components/Tabs/TabList.web.tsx (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/src/components/Tabs/TabList.web.tsx]: Updated

**Review by saurabhdaware — APPROVED** [review]

**Pull request merged**